### ISSUE TITLE
Add page with CLI & JSON/RPC examples

### ISSUE THREAD

**Issue by michaelherger** [LMS-Community/lms-community.github.io#5]
We have a pretty complete [CLI documentation built in to LMS](http://htmlpreview.github.io/?https://raw.githubusercontent.com/Logitech/slimserver/HEAD/HTML/EN/html/docs/cli-api.html). But it doesn't cover the use of those commands using tools like netcat, curl, or wget.

**Comment by terual** [comment]
Good idea! If we also make a markdown version, how do we maintain both versions and keep them in sync?

**Comment by terual** [comment]
BTW, under the jsonrpc.js section in the docs there is an curl example, so it is really not too bad!

**Comment by terual** [comment]
Or is your idea to rewrite every example in the CLI documentation to curl instead of telnet?

**Comment by michaelherger** [comment]
I tried to use `html2md` with the document. But the outcome isn't great: all the navigation is missing, the code blocks are turned into quotes (due the poor markup in the document) etc. I'd rather leave it as it is, with a link or instructions online.

What I'd really like to see is a page with a few examples, eg. how to skip to the next track using netcat/curl, how to get player state etc. All the details about the CLI can stay where they are. Duplication would kill the effort. We'd rather try to clean up the existing file so it could be converted and updated online automatically.

**Comment by sodface** [comment]
I started to put some example commands together before noticing you already added a reference page, which seems nice and covers everything I was going to do (curl, wget, nc examples). What do you think it's missing?

The only extra thing I was thinking about mentioning was the difference in behavior of the various `nc` implementations, though I'm not sure I fully understand the nuances enough to correctly explain it in writing. For example, busybox `nc` always exits when feeding commands to stdin, often before the reply is received and written to stdout, so doing something like this can help if you are trying to capture the response:
```
{ printf "c0:25:a5:ab:cd:ef status - 10\n"; sleep .1; } | busybox nc 192.168.1.1 9090
```
And openbsd netcat does sort of the opposite, staying connected after the command when you probably don't want it to. Adding `-N` helps here:
```
printf "c0:25:a5:ab:cd:ef playlist index +1\n" | nc -N 192.168.1.1 9090
```
Not sure if that's worth wading into or not?

Also, the IP in this example is missing an octet:

https://github.com/LMS-Community/lms-community.github.io/blob/5308de908008684e6dd7d43f18d495c8001e38af/docs/reference/using-the-cli.md?plain=1#L32

**Comment by michaelherger** [comment]
I believe the busybox netcat is a special case, as most of the busybox tools come with a reduced/different feature set. We should not really handle this here, but probably leave a note saying that it can be different, in particular on pCP.

**Comment by michaelherger** [comment]
Oh, and thanks for the heads up about the IP address. Should be fixed shortly.

**Comment by terual** [comment]
I guess now that Daksol's PR (https://github.com/LMS-Community/lms-community.github.io/pull/24) is merged this can be closed. Thanks @Daksol!

**Comment by Daksol** [comment]
# Note on migration of the CLI related material.

Source was LMS version 8.5

https://github.com/LMS-Community/slimserver/blob/public/8.5/HTML/EN/html/docs/cli-api.html

## Migrated content:- Target was this set of files in the lyrion.org repository

- [Introduction](https://lyrion.org/reference/cli/introduction/)
- [Using the CLI](https://lyrion.org/reference/cli/using-the-cli/)
- [General](https://lyrion.org/reference/cli/general/): general management of the Command Line Interface and of the server.
- [Players](https://lyrion.org/reference/cli/players/): management of players.
- [Database](https://lyrion.org/reference/cli/database/): mangement of the music database.
- [Playlist](https://lyrion.org/reference/cli/playlist/): management of the playlist of each player.
- [Coumpound Queries](https://lyrion.org/reference/cli/compoundqueries/): queries to get most of the information about the server or a player in one convenient query, that can be updated by the server automatically.
- [Notifications](https://lyrion.org/reference/cli/notifications/): internal server commands echoed to the CLI by using the "listen" or "subscribe" commands.
- [Alarms](https://lyrion.org/reference/cli/alarms/): management of alarms.

## Not migrated content

Two chunks of content from the original file were NOT migrated. For clarity the html of those sections included in this comment as zipped html attachments:-
- Plugins - cli-api-plugins.html.zip  [cli-api-plugins.html.zip](https://github.com/LMS-Community/lms-community.github.io/files/14883487/cli-api-plugins.html.zip)
- Deprecated cli-api-deprecated.html.zip  [cli-api-deprecated.html.zip](https://github.com/LMS-Community/lms-community.github.io/files/14883486/cli-api-deprecated.html.zip)
